### ISSUE TITLE
Crashes with "Invalid Version: 3.x"

### ISSUE THREAD

**Issue by jaredbeck** [jdanil/libyear#83]
# 🐛 Bug Report

`yarn libyear` crashes with mysterious output "Invalid Version: 3.x"

## 🎛 Configuration

No config

## 🤔 Expected Behavior

`libyear` should not crash. Perhaps the invalid dependency should be excluded from calculations?

## 😯 Current Behavior

```plain
yarn libyear          
Invalid Version: 3.x
✨  Done in 19.38s.
```

## Analysis

I added some logging, and *something* has a dependency on "svelte 3.x"

```plain
dependency:  svelte 3.x
begin getReleasesByType [
  '3.x',    '0.0.1',  '0.0.2',  # <------ A HA!
 ...
] major
Trace
    at Object.getReleasesByType (/redacted/node_modules/libyear/src/versions.js:11:13)
    at /redacted/node_modules/libyear/src/libyear.js:27:34
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Promise.all (index 89)
    at async Object.cli (/redacted/node_modules/libyear/src/cli.js:34:24)
Invalid Version: 3.x
```

In `/redacted/node_modules/swiper/package.json` I find:

```
  "peerDependencies": {
    "svelte": "^3.x"
  }
```

This seems a likely suspect. Perhaps we can catch this error and exclude swiper or svelte from the calculations?

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| libyear          | 0.6.2           |
| Node             | v12.22.3           |
| Package Manager  |  yarn 1.22.10           |

**Comment by jdanil** [comment]
Hi @jaredbeck 👋 

Thanks for taking the time to raise and investigate this issue. I've just pushed a commit for this (bf77560c4b6174d0352ad79f500a66bd5e12998a) so it should be resolved in `0.7.0`. Let me know if you still face any trouble.

The problem with yarn v1 is that it doesn't expose any CLI commands to fetch the direct dependencies of a project, so at the moment we're relying on `npm ls`. This can be sensitive to issues with unmet peer dependencies, which I suspect is what's happening here. Otherwise it really shouldn't try to parse peer dependencies at all.

As a workaround for now, I've decided to filter out these invalid versions as you suggested. Especially in this case, it shouldn't have been parsing peer dependencies anyway, so there is no real loss. I considered replacing the wildcard `x` versions with `0`, which would probably work most of the time, but would fall down in cases where a `0` version was skipped from being published to the registry for whatever reason.

I haven't dug too deep into finding a better solution for the peer dependencies error for yarn v1 as yarn v2 introduced a new CLI command that `libyear` uses, which is not susceptible to this issue.

**Comment by jaredbeck** [comment]
After upgrading to `0.7.0`, I can no longer reproduce this crash. Thanks, Josh!

Testing note: Swiper has since [removed Svelte from its peerDependencies](https://github.com/nolimits4web/swiper/blob/164e91ceaa1285b9d91c96f5c26b6f63583bee27/CHANGELOG.md#655-2021-04-08) so I had to downgrade Swiper to 6.5.4 to test this issue.
